cmsl1t/plotting/resolution.py

```diff
@@ -0,0 +1,104 @@
+        self.pileup_bins = bn.Sorted(pileup_bins, "pileup",
+                                     use_everything_bin=True)
+
+        name = ["resolution", self.online_name, self.offline_name, "pu_{pileup}"]
+        name = "__".join(name)
+        title = " ".join([self.online_name, "vs.", self.offline_name, "in PU bin: {pileup}"])
+        title = ";".join([title, self.offline_title, self.online_title])
+        self.plots = HistogramCollection([self.pileup_bins],
+                                         "Hist1D", n_bins, low, high,
+                                         name=name, title=title)
+        self.filename_format = name
+
+    def fill(self, pileup, offline, online):
+        difference = self.resolution_method(online, offline)
+        self.plots[pileup].fill(difference)
+
+    def draw(self, with_fits=False):
+        hists = []
+        labels = []
+        fits = []
+        for (pile_up, ), hist in self.plots.flat_items_all():
+            if pile_up == bn.Base.everything:
+                hist.linestyle = "dashed"
+                label = "Everything"
+            elif isinstance(pile_up, int):
+                hist.drawstyle = "EP"
+                label = "~ {:.0f}".format(self.pileup_bins.get_bin_center(pile_up))
+            else:
+                continue
+            hists.append(hist)
+            labels.append(label)
+            # if with_fits:
+            #     fits.append(self.fits.get_bin_contents([pile_up]))
+        self.__make_overlay(hists, fits, labels, "Number of events")
+
+        normed_hists = [hist / hist.integral() if hist.integral() != 0 else hist.Clone() for hist in hists]
+        self.__make_overlay(normed_hists, fits, labels, "Fraction of events", "__shapes")
+
+    def __make_overlay(self, hists, fits, labels, ytitle, suffix=""):
+        with preserve_current_style():
+            # Draw each resolution (with fit)
+            xtitle = self.resolution_method.label.format(on=self.online_title, off=self.offline_title)
+            canvas = draw(hists, draw_args={"xtitle": xtitle, "ytitle": ytitle})
+            if fits:
+                for fit, hist in zip(fits, hists):
+                    fit["asymmetric"].linecolor = hist.GetLineColor()
+                    fit["asymmetric"].Draw("same")
+
+            # Add labels
+            label_canvas()
+
+            # Add a legend
+            legend = Legend(len(hists), header="Pile-up bin",
+                            topmargin=0.35, entryheight=0.035)
+            for hist, label in zip(hists, labels):
+                legend.AddEntry(hist, label)
+            legend.Draw()
+
+            # Save canvas to file
+            name = self.filename_format.format(pileup="all")
+            self.save_canvas(canvas, name + suffix)
+
+    def _is_consistent(self, new):
+        """
+        Check the two plotters are the consistent, so same binning and same axis names
+        """
+        return (self.pileup_bins.bins == new.pileup_bins.bins) and \
```

Comment:
same comment as for the previous PR
&rarr; use `all`

Reply:
These files were added in #76, I'm not sure why this hasn't updated the diff, but I'll merge `weekly-checks` back into this and push to update things.

Reply:
Merged in the updated `weekly-checks` branch to this and now the diff looks as I'd expected.